superset-frontend/src/SqlLab/components/SqlEditor/index.jsx

```diff
@@ -592,6 +594,12 @@ class SqlEditor extends React.PureComponent {
     );
   }
 
+  async saveQuery(query) {
+    const { queryEditor: qe, actions } = this.props;
+    const savedQuery = await actions.saveQuery(query);
+    actions.addSavedQueryToTabState(qe, savedQuery);
```

Comment:
@betodealmeida we need to return it so that we can used the savedQuery context here, to pass it into the next action.